gtsfm/frontend/detector_descriptor/superpoint.py

```diff
@@ -31,30 +31,45 @@ class SuperPointDetectorDescriptor(DetectorDescriptorBase):
     """Superpoint Detector+Descriptor implementation."""
 
     def __init__(
-        self, max_keypoints: int = 5000, use_cuda: bool = True, weights_path: Union[Path, str] = MODEL_WEIGHTS_PATH
+        self,
+        max_keypoints: int = 5000,
+        use_cuda: bool = True,
+        weights_path: Union[Path, str] = MODEL_WEIGHTS_PATH,
+        init_model_in_constructor: bool = False,
     ) -> None:
         """Configures the object.
 
         Args:
             max_keypoints: max keypoints to detect in an image.
             use_cuda (optional): flag controlling the use of GPUs via CUDA. Defaults to True.
             weights_path (optional): Path to the model weights. Defaults to MODEL_WEIGHT_PATH.
+            init_model_in_constructor: flag to control initialization of weights.
         """
         super().__init__(max_keypoints=max_keypoints)
         self._use_cuda = use_cuda and torch.cuda.is_available()
         self._config = {"weights_path": weights_path}
 
+        # TODO: do not merge to master. Models should not be initialized in constructors for dask.
```

Comment:
What do we do here?

Reply:
Removed this comment. We have introduced a constructor param to configure loading of models